superset-frontend/src/components/UnsavedChangesModal/UnsavedChangesModal.test.tsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { fireEvent, render, waitFor } from 'spec/helpers/testing-library';
+import UnsavedChangesModal from '.';
+
+test('should render nothing if showModal is false', () => {
+  const { queryByTestId } = render(
+    <UnsavedChangesModal
+      showModal={false}
+      onHide={() => {}}
+      handleSave={() => {}}
+      onConfirmNavigation={() => {}}
+    />,
+  );
+
+  expect(queryByTestId('unsaved-changes-modal')).not.toBeInTheDocument();
+});
+
+test('should render the UnsavedChangesModal component if showModal is true', async () => {
+  const { findByTestId } = render(
+    <UnsavedChangesModal
+      showModal
+      onHide={() => {}}
+      handleSave={() => {}}
+      onConfirmNavigation={() => {}}
+    />,
+  );
+
+  expect(await findByTestId('unsaved-changes-modal')).toBeInTheDocument();
```

Comment:
Can we rely on roles whenever possible? https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices#keep-accessibility-in-mind-when-writing-your-tests